### PR TITLE
[RFR] Overhaul IPAppliance/Appliance interface

### PR DESCRIPTION
**Pull request by mfalesni**
@psav @RonnyPfannschmidt 
{{pytest: -v -m smoke --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/utils/appliance/db.py [hunk 1]
```diff
@@ -371,7 +371,7 @@ def is_enabled(self):
     @property
     def is_internal(self):
         """Is database internal"""
-        if self.address == self.appliance.address:
+        if self.address == self.appliance.hostname:
```
Comment (RonnyPfannschmidt) [inline cfme/utils/appliance/db.py]: **optional** just return the boolean directly

**Comment by mfalesni** [comment]
The last test results before rebase looked okay. Please review.

**Comment by mfalesni** [comment]
Results are okay, only ec2west crud failed, if this was wrong, it would be failing hard.

**Comment by jkrocil** [comment]
Looks good, checked it locally as well. :+1:  Merging

**Pull request merged**